src/cmap/wire_protocol/on_demand/document.ts

```diff
@@ -45,6 +46,13 @@ type CachedBSONElement = { element: BSONElement; value: any | undefined };
 
 /** @internal */
 export class OnDemandDocument {
+  /**
+   * @internal
+   *
+   * Used for testing purposes.
+   */
+  private static BSON: typeof BSON = BSON;
```

Comment:
Why do we need to do this but not something similar for `net`?

Reply:
Let's determine the reason before introducing this injectable. Are you able to debug? I will take a look when I'm at my desk